packages/flame/lib/src/components/isometric_tile_map_component.dart

```diff
@@ -75,20 +82,36 @@ class IsometricTileMapComponent extends PositionComponent {
   /// This is the size the tiles will be drawn (either original or overwritten).
   Vector2 get effectiveTileSize => destTileSize ?? tileset.srcSize;
 
-  /// This is the vertical height of each block; by default it's half the tile
-  /// size.
+  /// Gets the current Scaling factor for the isometric View
+  double get scalingFactor => effectiveTileSize.y / effectiveTileSize.x;
+
+  /// This is the vertical height of each block; by default it's half the tile size.
   double get effectiveTileHeight => tileHeight ?? (effectiveTileSize.y / 2);
 
+  /// gets the effective propSize. PropSize should be assigned if prop dimension are different from srcSize of tilset
+  Vector2 get pSize => propSize ?? effectiveTileSize;
+
   @override
   void render(Canvas c) {
     final size = effectiveTileSize;
     for (var i = 0; i < matrix.length; i++) {
       for (var j = 0; j < matrix[i].length; j++) {
         final element = matrix[i][j];
         if (element != -1) {
-          final sprite = tileset.getSpriteById(element);
-          final p = getBlockRenderPositionInts(j, i);
-          sprite.render(c, position: p, size: size);
+          final Sprite currentSprite;
+          if (element < -1) {
+            final yRowShift = ((element + 2).abs() / (tileset.image.width / pSize.x).floor()).floor() + 1;
+            final xColShift = (element + 2).abs() % (tileset.image.width / pSize.x).floor();
+            final sourcePosition = Vector2(
```

Comment:
We shouldn't create a new `Vector2` object in every tick here, you can create a private vector outside of the render method that you use instead, for example `_renderSourcePosition` and then you use it in here with `_renderSourcePosition.setValues`.